File: SchoolWebApp-API/Project.Core/Entities/School/SchoolStream.cs
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.School
{
    public class SchoolStream: Base
    {
        public required string Name { get; set; }
        public string? Description { get; set; }
        public required string Abbreviation { get; set; }
        public int Rank { get; set; }
        public List<SchoolClass> SchoolClasses { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/class/models/school-stream.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class SchoolStream extends ResourceModel<SchoolStream> {
    public name?: string;
    public description?: string;
    public abbreviation?: string;
    public rank?: number;
    
    constructor(model?: Partial<SchoolStream>) {
        super(model);
    }
}
